### PR TITLE
Pilot sections docs

### PR DESCRIPTION
**Pull request by matyasselmeci**
SOFTWARE-4430

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -458,6 +458,41 @@ The following attributes are optional:
...
+For example, if your CE is registered in topology as `UW-ITB-CE1` and you are describing a type of pilot with 4 cores,
+then use `UW-ITB-CE1_4CORE`.
+
+The following attributes are required:
+| Option                   | Values Accepted             | Explanation                                                                                                                        |
+|--------------------------|-----------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| **allowed\_vos**         | Comma-separated List or `*` | The VOs that are allowed to run jobs on this resource (autodetected if `*`)                                                        |
+| **max\_pilots**          | Positive Integer            | The maximum number of pilots of this type that the factory can send to this CE                                                     |
+| **os**                   | Choice (see below)          | The OS on the workers the pilot should request.  Not set by default.  Only required if **require\_singularity** is `False`               |
```
Comment (brianhlin) [inline README.md:475]:
```suggestion
| **os**                   | Choice (see below)          | The operating system on the workers the pilot should request.  Not set by default.  Only required if **require\_singularity** is `False`               |
```

File: config/35-pilot.ini [hunk 2]
```diff
@@ -3,15 +3,15 @@
 ;===================================================================
 
 ; For each pilot type, add a new pilot section.
-; Each pilot name must be unique for the entire grid, so make sure to not
-; pick anything generic like "MAIN". The name will be used as-is as the
+; The name of each pilot type must be unique for the entire grid, so make sure
```
Comment (brianhlin) [inline config/35-pilot.ini:6]: Why must it be unique? And it would be great if we could provide a little more guidance on this

File: README.md [hunk 3]
```diff
@@ -458,6 +458,41 @@ The following attributes are optional:
 
 
 
+### 35-pilot.ini / [Pilot] ###
+
+These sections describe the types of glideinWMS pilots that your site is willing to accept.
```
Comment (brianhlin) [inline README.md:463]:
```suggestion
These sections describe the size and scale of GlideinWMS pilots that your site is willing to accept.
```

Or something to that effect. Also, GlideinWMS is the official capitalization according to https://glideinwms.fnal.gov/doc.prd/install.html

**Review by brianhlin — APPROVED** [review]
Two minor comments but otherwise LGTM

File: README.md (hunk 3 above)
Comment (matyasselmeci) [reply README.md:463]: thanks, I was looking for something more descriptive than 'types' but couldn't think of it.

File: config/35-pilot.ini (hunk 2 above)
Comment (matyasselmeci) [reply config/35-pilot.ini:6]: It's used in the `Name` attribute of the `OSG_ResourceCatalog` attribute that gets sent by the CE to the central collector -- just like `Subcluster` and `Resource Entry` sections. It's not absolutely necessary for it to be unique -- ads from different CEs won't stomp on each other -- but uniqueness makes it easier to query.

The advice I recommended in the doc is that the name should start with the name of the CE's topology resource; I can update the comment to say the same.

We could also make osg-configure prepend the resource name to the pilot type name (so if you have `resource=ITB-CE1` in the `Site Information` section, and name this section `Pilot 4CPU`, then the Name attribute in the OSG_ResourceCatalog entry would be `ITB-CE1-4CPU`. I can't remember if we discussed that or not...

Comment (brianhlin) [reply config/35-pilot.ini:6]: I vote that we remove the global requirement for `Pilot` and whatever the non-ATLAS one is between `Subcluster` and `Resource Entry` and just insist that they're unique among entries for that CE. For our own queries, it's easy enough to find unique entries based on the combination of `OSG_Resource` and pilot entry name in the schedd ad. It's much harder for users to make sure their name is unique across the entire OSG.

I think concatenating the resource and pilot entry names from the config to create the advertised pilot name is nice but unnecessary given the above.

Comment (matyasselmeci) [reply config/35-pilot.ini:6]: OK. (I think the global uniqueness constraint was for BDII back in the day and we just kept it because we keep things.)

Comment (matyasselmeci) [reply config/35-pilot.ini]: Updated.

**Review by brianhlin — APPROVED** [review]
LGTM

**Pull request merged**